src/main/java/com/multiplex/booking/BookingController.java

```diff
@@ -0,0 +1,279 @@
+        Duration duration = Duration.between(LocalDateTime.now(),showing.getShowingTime());
+        long minutes = duration.toMinutes();
+
+        if(minutes <= 15) throw new CannotReserveException();
+
+        Set<Seat> seats = showing.getSeats();
+
+        double total = 0;
+
+        for (Seat newSeat : newSeats) {
+            Optional<Seat> seatOptional = seats.stream()
+                .filter(seat -> seat.getId().equals(newSeat.getId()))
+                .findFirst();
+
+            if (seatOptional.isPresent()) {
+                Seat seat = seatOptional.get();
+
+                String name = newSeat.getName();
+                String surname = newSeat.getSurname();
+                String ticketType = newSeat.getTicketType();
+
+                // seat cannot be reserved, name, surname and ticket type is required
+                if(seat.isReserved()) throw new WrongInputException();
+                if(name == null) throw new WrongInputException();
+                if(surname == null) throw new WrongInputException();
+                if(ticketType == null) throw new WrongInputException();
+
+                // name must start with an uppercase
+                if(!Character.isUpperCase(name.charAt(0))) throw new WrongInputException();
+
+                // surname must start with an uppercase and could consist of two parts separated with
+                // a single dash, in this case the second part should also start with a capital letter
+                if(surname.contains("-")){
+                    String[] surname_parts = surname.split("-");
+                    String first_surname = surname_parts[0];
+                    String second_surname = surname_parts[1];
+                    if(!Character.isUpperCase(first_surname.charAt(0))) throw new WrongInputException();
+                    if(!Character.isUpperCase(second_surname.charAt(0))) throw new WrongInputException();
+                }
+                else{
+                    if(!Character.isUpperCase(surname.charAt(0))) throw new WrongInputException();
+                }
+
+                seat.setReserved(true);
+                seat.setName(name);
+                seat.setSurname(surname);
+                seat.setTicketType(ticketType);
+
+                if(newSeat.getTicketType().equals("adult")) total += ADULT_PRICE;
+                if(newSeat.getTicketType().equals("student")) total += STUDENT_PRICE;
+                if(newSeat.getTicketType().equals("child")) total += CHILD_PRICE;
+
+            } else {
+                throw new SeatNotFoundException(newSeat.getId());
+            }
+        }
+
+        if(!showing.isValid()) throw new CannotReserveException();
+
+        showings.save(showing);
+        
+        ReservationResponse response = new ReservationResponse();
+        response.setTotal(total);
+        response.setReservationExpirationTime(showing.getShowingTime().minusMinutes(15));
```

Comment:
Czy musimy używać tutaj setterów? Czy moglibyśmy użyć konstruktora?

Reply:
Wystarczyło by dopisać konstruktor, który przyjmował by oba parametry:
```
public ReservationResponse(double total, LocalDateTime reservationExpirationTime) {
    this.total = total;
    this.reservationExpirationTime = reservationExpirationTime;
}
```
Następnie mogłabym użyć go zamiast setterów:

```suggestion
        ReservationResponse response = new ReservationResponse(total, showing.getShowingTime().minusMinutes(15));
```